sql/expression/function/json/json_quote_test.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package json
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/expression"
+	"github.com/dolthub/go-mysql-server/sql/types"
+)
+
+func TestJSONQuote(t *testing.T) {
+	testCases := []struct {
+		arg sql.Expression
+		exp interface{}
+		err bool
+	}{
+		{
+			arg: expression.NewLiteral(true, types.Boolean),
+			err: true,
+		},
+		{
+			arg: expression.NewLiteral(123, types.Int64),
+			err: true,
+		},
+		{
+			arg: expression.NewLiteral(123.0, types.Float64),
+			err: true,
+		},
+		{
+			arg: expression.NewLiteral(types.MustJSON(`{"a": 1}`), types.JSON),
+			err: true,
+		},
+		{
+			arg: expression.NewLiteral(nil, types.Null),
+			exp: nil,
+		},
+		{
+			arg: expression.NewLiteral("abc", types.Text),
+			exp: `"abc"`,
+		},
+		{
+			arg: expression.NewLiteral(`[1, 2, 3]`, types.Text),
+			exp: `"[1, 2, 3]"`,
+		},
+		{
+			arg: expression.NewLiteral(`"\t\u0032"`, types.Text),
+			exp: `"\"\\t\\u0032\""`,
+		},
+		{
+			arg: expression.NewLiteral(`\`, types.Text),
+			exp: `"\\"`,
+		},
+		{
+			arg: expression.NewLiteral(`\b\f\n\r\t\"`, types.Text),
+			exp: `"\\b\\f\\n\\r\\t\\\""`,
```

Comment:
For this one, with MySQL, I'm getting:
```sql
mysql> select JSON_QUOTE('\b\f\n\r\t\"');
+----------------------------+
| JSON_QUOTE('\b\f\n\r\t\"') |
+----------------------------+
| "\bf\n\r\t\""              |
+----------------------------+
```

The MySQL docs call out that `\f` is a supported escape sequence, but it also says for non-supported escape sequences it ignores the `\`, which seems to be what it's doing here with `\f`.

Reply:
Same as above
```sql
tmp/main> select json_quote("\b\f\n\r\t\"");
+----------------------------+
| json_quote("\b\f\n\r\t\"") |
+----------------------------+
| "\bf\n\r\t\""              |
+----------------------------+
1 row in set (0.00 sec)
```